core/trino-main/src/main/java/io/trino/execution/TaskManagerConfig.java

```diff
@@ -69,6 +69,7 @@
 
     private Duration statusRefreshMaxWait = new Duration(1, TimeUnit.SECONDS);
     private Duration infoUpdateInterval = new Duration(3, TimeUnit.SECONDS);
+    private Duration interruptRunawaySplitsTimeout = new Duration(600, TimeUnit.SECONDS);
```

Comment:
We also have a warning about the runaway splits, currently hardcoded. IMO we should validate the two thresholds are coherent (i.e. warning <= timeout), which would require  making the warning configurable too

Reply:
got it, will configure: "LONG_SPLIT_WARNING_THRESHOLD"

Reply:
agree we should make  warning <= timeout, because if the thread runtime for the split > than timeout, the system will interrupt the split.

Reply:
I also felt "/v1/maxActiveSplits" is unnecessary.. Since this PR will print out the run away splits in the log..